rsocket-core/src/main/java/io/rsocket/metadata/TaggingMetadata.java

```diff
@@ -0,0 +1,88 @@
+package io.rsocket.metadata;
+
+import static io.netty.buffer.Unpooled.buffer;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.CompositeByteBuf;
+import io.netty.buffer.Unpooled;
+import java.nio.charset.StandardCharsets;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.Spliterator;
+import java.util.Spliterators;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+/**
+ * Tagging metadata from https://github.com/rsocket/rsocket/blob/master/Extensions/Routing.md
+ *
+ * @author linux_china
+ */
+public class TaggingMetadata implements Iterable<String>, CompositeMetadata.Entry {
+  /** Tag max length in bytes */
+  private static int TAG_LENGTH_MAX = 0xFF;
+
+  private String type;
+  private ByteBuf content;
+
+  public TaggingMetadata(String type, ByteBuf content) {
+    this.type = type;
+    this.content = content;
+  }
+
+  public TaggingMetadata(String type, Collection<String> tags) {
+    this.type = type;
+    CompositeByteBuf compositeByteBuf = Unpooled.compositeBuffer();
```

Comment:
`CompositeMetadataFlyweight` allows any `ByteBufAllocator` to be passed in while here it is fixed.